test/Tester/MethodInterceptionTests.cs

```diff
@@ -25,5 +29,38 @@ public class MethodInterceptionTests : HostedTestClusterEnsureDefaultStarted
             result = await grain.SayHello();
             Assert.Equal("Hello", result);
         }
+
+        [Fact]
+        public async Task GenericGrainMethodInterceptionTest()
+        {
+            var grain = GrainFactory.GetGrain<IGenericMethodInterceptionGrain<int>>(
+                0,
+                typeof(GenericMethodInterceptionGrain<>).Namespace + "." + nameof(GenericMethodInterceptionGrain<int>));
```

Comment:
This seems very cryptic and unusable.  How would a user ever know how to form this sort of magic?
Also required user know about concrete type (at least it's name).
Are we confident there is not a good way around requiring this hint?

If we're going to require something like this can we at least add some  IGrainFactory extensions for this, something like:
GetGenericGrain<TInterface,TGrainType>(grainId)

If this is out of scope for this PR, we can address this in a separate PR, or open an issue.